File: 242-valid-anagram/valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        check = {}
        if(len(s) != len(t)):
            return False
        for char in s:
            if char in check:
                check[char] += 1
            else:
                check[char] = 1
        
        for char in t:
            if char not in check or check[char] == 0:
                return False
            check[char] -= 1
        return True
    
        

File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> counter;
        for (char c : s){
            counter[c]++;
        }
        for (char c : t){
            counter[c]--;
        }
        for (auto it : counter){
            if(it.second != 0) return false;
        }
        return true;
    }
};